### PR TITLE
verifying spanner endpoint before execution to avoid the client api calls getting stuck

### PR DESCRIPTION
**Pull request by shreyakhajanchi**
In case of incorrect spanner api endpoint, the client APIs get stuck as in the background they have a timeout of 60 min. In order to inform the user about the probable incorrect endpoint added a warning to be printed at the start of the process.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -100,6 +100,15 @@ func main() {
 	}
 	defer utils.Close(lf)
 
+	endpoint := os.Getenv("SPANNER_API_ENDPOINT")
+	emulator_host := os.Getenv("SPANNER_EMULATOR_HOST")
+	if !((endpoint == "" && (emulator_host == "" || emulator_host == "localhost:9010")) || (endpoint == "staging-wrenchworks.sandbox.googleapis.com:443" && emulator_host == "")) {
```
Comment (bharadwaj-aditya) [inline main.go:105]: Please correct this naming error

File: main.go [hunk 2]
```diff
@@ -100,6 +100,15 @@ func main() {
...
+		fmt.Printf("\nWarning: Endpoint specified may be incorrect: \n" +
+			"For connecting to prod set SPANNER_API_ENDPOINT as blank \n" +
+			"For connecting to emulator set SPANNER_EMULATOR_HOST=localhost:9010 \n" +
+			"For connecting to staging set SPANNER_API_ENDPOINT=staging-wrenchworks.sandbox.googleapis.com:443\n\n")
```
Comment (bharadwaj-aditya) [inline main.go:109]: It is a good idea to use constants for strings like this, so that if we change it in one place, it reflects everywhere.

**Review by bharadwaj-aditya — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (shreyakhajanchi) [reply main.go:105]: corrected.

File: main.go (hunk 2 above)
Comment (shreyakhajanchi) [reply main.go:109]: added constants

File: main.go [hunk 3]
```diff
@@ -100,6 +100,15 @@ func main() {
 	}
 	defer utils.Close(lf)
 
+	endpoint := os.Getenv("SPANNER_API_ENDPOINT")
+	emulatorHost := os.Getenv("SPANNER_EMULATOR_HOST")
+	if !((endpoint == "" && (emulatorHost == "" || emulatorHost == constants.SPANNER_EMULATOR_HOST)) || (endpoint == constants.STAGING_SPANNER_API_ENDPOINT && emulatorHost == "")) {
```
Comment (Deep1998) [inline main.go:105]: This if condition is difficult to read. Suggest moving this to a 
func checkSpannerEndpoints() bool {
   endpoint := os.Getenv("SPANNER_API_ENDPOINT")
   emulatorHost := os.Getenv("SPANNER_EMULATOR_HOST")
  // list down the conditions separately.
}

File: main.go [hunk 4]
```diff
@@ -100,6 +100,15 @@ func main() {
...
+		fmt.Printf("\nWarning: Endpoint specified may be incorrect: \n" +
+			"For connecting to prod set SPANNER_API_ENDPOINT as blank \n" +
+			"For connecting to emulator set SPANNER_EMULATOR_HOST=" + constants.SPANNER_EMULATOR_HOST + "\n" +
+			"For connecting to staging set SPANNER_API_ENDPOINT=" + constants.STAGING_SPANNER_API_ENDPOINT + "\n\n")
```
Comment (Deep1998) [inline main.go:109]: Do we want to print this staging url to all the users?

File: main.go [hunk 5]
```diff
@@ -100,6 +100,15 @@ func main() {
 	}
 	defer utils.Close(lf)
 
+	endpoint := os.Getenv("SPANNER_API_ENDPOINT")
+	emulatorHost := os.Getenv("SPANNER_EMULATOR_HOST")
+	if !((endpoint == "" && (emulatorHost == "" || emulatorHost == constants.SPANNER_EMULATOR_HOST)) || (endpoint == constants.STAGING_SPANNER_API_ENDPOINT && emulatorHost == "")) {
+		fmt.Printf("\nWarning: Endpoint specified may be incorrect: \n" +
+			"For connecting to prod set SPANNER_API_ENDPOINT as blank \n" +
```
Comment (Deep1998) [inline main.go:107]: Would it be clear to the user that we are talking about the env variable SPANNER_API_ENDPOINT?
Also suggest fixing the punctuation a bit.
```
fmt.Printf("\nWarning: Endpoint specified may be incorrect\n" +
			"For connecting to prod, run: export SPANNER_API_ENDPOINT=\"\" \n" +
			"For connecting to emulator, run: export SPANNER_EMULATOR_HOST=" + constants.SPANNER_EMULATOR_HOST + "\n" +
			"For connecting to staging, run: export SPANNER_API_ENDPOINT=" + constants.STAGING_SPANNER_API_ENDPOINT + "\n\n")
```

**Pull request closed without merging**